### PR TITLE
Global schema lock basically usable completed

### PR DESCRIPTION
**Pull request by soulasuna**
## Global schema lock basically usable completed

Related to #16269.

Changes proposed in this pull request:
- Global schema lock basically usable completed.

### CODE REVIEW

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/test/java/org/apache/shardingsphere/spring/namespace/fixture/FixtureClusterPersistRepository.java [hunk 1]
```diff
@@ -54,8 +54,9 @@ public void persist(final String key, final String value) {
     }
     
     @Override
-    public void persistEphemeral(final String key, final String value) {
+    public boolean persistEphemeral(final String key, final String value) {
```
Comment (terrymanu) [inline shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/test/java/org/apache/shardingsphere/spring/namespace/fixture/FixtureClusterPersistRepository.java:57]: It is better to return `void` value here, and catch exception in `tryLock` to return boolean value.

File: shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/future/lock/service/GlobalLockRegistryService.java [hunk 2]
```diff
@@ -40,7 +58,7 @@ public GlobalLockRegistryService(final ClusterPersistRepository repository) {
      * @return true if get the lock, false if not
      */
     public boolean tryLock(final String lockName, final long timeoutMilliseconds) {
-        return repository.tryLock(lockName, timeoutMilliseconds, TimeUnit.MILLISECONDS);
+        return repository.persistEphemeral(lockName, LockAck.LOCKED.name());
```
Comment (terrymanu) [inline shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/future/lock/service/GlobalLockRegistryService.java:61]: Just catch exception here, it is better to keep original return type with `persistEphemeral`

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/test/java/org/apache/shardingsphere/spring/namespace/fixture/FixtureClusterPersistRepository.java (hunk 1 above)
Comment (soulasuna) [reply shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/test/java/org/apache/shardingsphere/spring/namespace/fixture/FixtureClusterPersistRepository.java:57]: OK, I will fix them.

File: shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/future/lock/service/GlobalLockRegistryService.java (hunk 2 above)
Comment (soulasuna) [reply shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/future/lock/service/GlobalLockRegistryService.java:61]: OK, I will fix them.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**